src/Api/Domain.php

```diff
@@ -88,16 +88,26 @@ public function create(string $domain, string $smtpPass = null, string $spamActi
 
         $params['name'] = $domain;
 
-        // If at least smtpPass available, check for the fields spamAction wildcard
         if (!empty($smtpPass)) {
+            Assert::stringNotEmpty($smtpPass);
+            Assert::lengthBetween($smtpPass, 5, 32, 'SMTP password must be between 5 and 32 characters.');
+
+            $params['smtp_password'] = $smtpPass;
+        }
+
+        if (null !== $spamAction) {
             // TODO(sean.johnson): Extended spam filter input validation.
             Assert::stringNotEmpty($spamAction);
-            Assert::boolean($wildcard);
 
-            $params['smtp_password'] = $smtpPass;
             $params['spam_action'] = $spamAction;
         }
 
+        if (null !== $wildcard) {
```

Comment:
👍 This one makes sense and there is no need to change it, due to the `!empty` check would not work properly for the `false` value